module/x/gravity/keeper/ethereum_event_handler.go

```diff
@@ -121,3 +78,102 @@ func (a EthereumEventProcessor) Handle(ctx sdk.Context, eve types.EthereumEvent)
 	}
 	return nil
 }
+
+func (a EthereumEventProcessor) verifyERC20DeployedEvent(ctx sdk.Context, event *types.ERC20DeployedEvent) error {
+	if existingERC20, exists := a.keeper.getCosmosOriginatedERC20(ctx, event.CosmosDenom); exists {
+		return sdkerrors.Wrapf(
+			types.ErrInvalid,
+			"ERC20 token %s already exists for denom %s", existingERC20.Hex(), event.CosmosDenom,
+		)
+	}
+
+	// We expect that all Cosmos-based tokens have metadata defined. In the case
+	// a token does not have metadata defined, e.g. an IBC token, we successfully
+	// handle the token under the following conditions:
+	//
+	// 1. The ERC20 name is equal to the token's denomination. Otherwise, this
+	// 		means that ERC20 tokens would have an untenable UX.
+	// 2. The ERC20 token has zero decimals as this is what we default to since
```

Comment:
Fede, I don't think that is a safe assumption. Also, there is no default. ATOM uses 6 and many tokens copy that but there's nothing stopping another zone using 18 or something else, and then this would break for them. 0 is the safest default IMO.

Reply:
we use 18 for Ethermint. But I think pretty much everyone else uses 6. I think leaving it as 0 is ok, as long as we raise the issue to add the data on ICS20

Reply:
>  as long as we raise the issue to add the data on ICS20

What does this mean? metadata exists in `x/bank` and is defined at genesis. Nothing to do with ICS20 or IBC AFAIK.